projects/plugins/jetpack/extensions/blocks/payments-intro/block.json

```diff
@@ -0,0 +1,21 @@
+{
```

Comment:
I don't know if we need this one, since that's just a parent block for the Payments button, Donations, and Paid content blocks?

Reply:
`jetpack/payments-intro` is a block like any other: it's registered with `register_block_type`, can be inserted from the inserter, and is serialized as `<!-- wp:jetpack/payments-intro /-->`. What is special about is that it's meant to be a temporary placeholder, to be quickly replaced by one of the "real" payment blocks. But that's just semantics, formally it's a block.

Reply:
Right, but I'm not sure it's one that we must publicize on W.org, given that we already publicize the "real" payment blocks. It may create more confusion than anything else.

Reply:
Having a `block.json` file should be useful and preferred for all blocks. For example, they are then properly visible in the `/block-types` REST endpoint. If we don't want to publicize them on W.org, maybe there is a `block.json` flag for that? @gziolo will know.

Reply:
You don't need `block.json` to expose the block in the REST endpoint. All you need is to register it with `register_block_type` so it's available in the block type registry.

I don't know precisely how W.org scans the code of the plugin when looking for block types. I bet though, that when you alias the function with something else, it won't list it.

Reply:
> All you need is to register it with `register_block_type` so it's available in the block type registry.

Jetpack currently doesn't do server-side block registration with the `block.json`. The subset of fields that it passes to `register_block_type` is a very small subset, and it's duplicated. So, the info available in the REST endpoint will be very limited. It would be a nice follow-up to improve this.

> I don't know precisely how W.org scans the code of the plugin when looking for block types.

That would be nice to know, especially if there is any "opt-out from public registry listing" flag. It would be very useful for cases like this one.